spec/services/work_combined_audio_creator_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe "Combined Audio" do
+  # We need to set the friendlier_id explicitly, because it's part of the fingerprint recipe.
+  let!(:work) { FactoryBot.create(:work, title: "top", friendlier_id: '1ciyp1i')}
+
+  let!(:audio_asset_1)  { create(:asset, :inline_promoted_file,
+      position: 1,
+      parent_id: work.id,
+      id: 'ddbd1a8d-c2eb-47b3-85a4-11b4ffb41719',
+      file: File.open((Rails.root + "spec/test_support/audio/ice_cubes.mp3"))
+    )
+  }
+
+  let!(:audio_asset_2)  { create(:asset, :inline_promoted_file,
+      position: 2,
+      parent_id: work.id,
+      id: 'df773502-56d7-4756-a58c-b1f479910e97',
+      file: File.open((Rails.root + "spec/test_support/audio/double_ice_cubes.mp3"))
+    )
+  }
+
+  it "creates combined audio derivatives" do
+    expect(work.members.map(&:stored?)).to match([true, true])
+    combined_audio_info = CombinedAudioDerivativeCreator.new(work).generate
+    expect(combined_audio_info.start_times.count).to eq 2
+    expect(combined_audio_info.start_times).to match([
+      ["ddbd1a8d-c2eb-47b3-85a4-11b4ffb41719", 0],
+      ["df773502-56d7-4756-a58c-b1f479910e97", 1.593469]
+    ])
+    cmd = TTY::Command.new(printer: :null)
+    stats_command = ['ffprobe', '-v', 'error', '-show_format', '-show_streams' ]
+    mp3_details   = cmd.run(*stats_command + [combined_audio_info.mp3_file.path] ).out.split("\n")
+    webm_details  = cmd.run(*stats_command + [combined_audio_info.webm_file.path]).out.split("\n")
+    expect(mp3_details).to  include('format_name=mp3')
+    expect(webm_details).to include('format_name=matroska,webm')
```

Comment:
I don't understand why you are you doing your own TTY::Command shell out in the test and testing that. Is this a mistake? It would appear to be testing code that is only located in this test, instead of testing the code that is in CombinedAudioDerivativeCreator . Please explain, if it isn't a mistake. 

Also, nothing in this test seems to be testing that the `Response` object returned actually includes the Tempfile for the webm and the mp3. Please ensure these are there, that they are ruby Tempfiles, and maybe that the file is actually non-0, or if you want to be extra that the file really looks like a webm/mp3 (maybe using the same file analysis tool that kithe uses, or shelling out to `file`, or whatever).

Reply:
So I'm using ffprobe here to determine that the files output by the service are in fact audio, that their duration is correct, and that they are indeed mp3 or webm. (I *could* use file, but ffprobe gives me duration in particular.)

Reply:
Definitely should be checking that `combined_audio_info.mp3_file.class == Tempfile`

Reply:
OH NOW I SEE

Reply:
OK -- resolving this!